src/charts/raster-chart.js

```diff
@@ -184,11 +184,20 @@ export default function rasterChart (parent, useMap, chartGroup, _mapboxgl) {
     const bounds = _chart.getDataRenderBounds()
     _chart._updateXAndYScales(bounds)
 
-    _chart._vegaSpec = genLayeredVega(_chart)
-
-    const nonce = _chart.con().renderVega(_chart.__dcFlag__, JSON.stringify(_chart._vegaSpec), {}, callbacks)
-
-    _renderBoundsMap[nonce] = bounds
+    const heatLayer = _chart.getLayer("heat")
```

Comment:
we update the raster chart `dataAsync` for the heat layer. we make an extra call to get the color domain for it's legend.